server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -579,7 +582,9 @@ public void writeTo(StreamOutput out) throws IOException {
                 out.writeMap(entry.userMetadata);
             }
             if (out.getVersion().onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)) {
```

Comment:
this should be `out.getVersion().onOrAfter(VERSION_IN_SNAPSHOT_VERSION)`?

Reply:
Right urgh ... we need a better BwC test for this stuff to show up in a deterministic fashion. I'll look into that.